msbuild/Xamarin.Mac.Tasks/Xamarin.Mac.Common.props

```diff
@@ -93,6 +93,26 @@ Copyright (C) 2013-2014 Xamarin. All rights reserved.
 				<MacBclPath>$(XamarinMacFrameworkRoot)/lib/mono/4.5</MacBclPath>
 			</PropertyGroup>
 		</When>
+		<When Condition=" '$(ForceNonSystemTargetFramework)' == 'true'">
+			<Choose>
+				<When Condition=" '$(TargetFrameworkVersion)' == '2.0' Or '$(TargetFrameworkVersion)' == ''">
+					<PropertyGroup>
+						<TargetFrameworkName>Modern</TargetFrameworkName>
+						<MacBclPath>$(XamarinMacFrameworkRoot)/lib/mono/Xamarin.Mac</MacBclPath>
+						<TargetFrameworkVersion>v2.0</TargetFrameworkVersion>
+						<TargetFrameworkIdentifier>Xamarin.Mac</TargetFrameworkIdentifier>
+					</PropertyGroup>
+				</When>
+				<Otherwise>
+					<PropertyGroup>
+						<TargetFrameworkName>Full</TargetFrameworkName>
+						<MacBclPath>$(XamarinMacFrameworkRoot)/lib/mono/4.5</MacBclPath>
+						<UseXamMacFullFramework>true</UseXamMacFullFramework>
+						<AssemblySearchPaths>$(MacBclPath);$(MacBclPath)/Facades</AssemblySearchPaths>
```

Comment:
In theory this is not needed, but it appears somewhere in the machinery before Xamarin.Mac.Common.props we don't set the AssemblySearchPaths path without UseXamMacFullFramework tag.

A very slight hack, but easier that restructuring the entire msbuild to get it set earlier.